magma/inline_verilog.py

```diff
@@ -94,7 +94,8 @@ def _insert_temporary_wires(
                     f"{repr(orig_value)}")
 
         key = value
-        if isinstance(value, ClockTypes) and not value.driven():
+        if (isinstance(value, ClockTypes) and value.is_input()
+                and not value.driven()):
```

Comment:
```suggestion
        is_undriven_clock = isinstance(value, ClockTypes) and value.is_input() and not value.driven()
        if is_undriven_clock:
```

Reply:
btw something I've found recently is that `value.driven() and value.trace() is None` can be true. e.g.

```python
x = m.Bit()
y = m.Bit()
y @= x
assert y.driven()  # driven by x
assert y.trace() is None  # x is not driven by anything
```

Reply:
yeah, agreed. we could change the name but as long as we have `.value()` and `.trace()` then we have all the info we need. seem like reasonable names